src/main/java/de/tum/in/www1/artemis/service/plagiarism/ContinuousPlagiarismControlService.java

```diff
@@ -64,12 +75,13 @@ public void executeChecks() {
         log.debug("Continuous plagiarism control done.");
     }
 
-    private void executeChecksForExercise(Exercise exercise) throws Exception {
-        switch (exercise.getExerciseType()) {
+    private PlagiarismResult<?> executeChecksForExercise(Exercise exercise) throws Exception {
+        return switch (exercise.getExerciseType()) {
             case TEXT -> plagiarismChecksService.checkTextExercise((TextExercise) exercise);
             case PROGRAMMING -> plagiarismChecksService.checkProgrammingExercise((ProgrammingExercise) exercise);
             case MODELING -> plagiarismChecksService.checkModelingExercise((ModelingExercise) exercise);
-            case FILE_UPLOAD, QUIZ -> log.error("Cannot check plagiarism for exercise: type={}, id={}.", exercise.getExerciseType(), exercise.getId());
-        }
+            case FILE_UPLOAD, QUIZ -> throw new IllegalStateException(
```

Comment:
Write an issue for creating a strategy for handling plagiarism cases in file upload exercises.

Reply:
#6921